src/rm_tdigest.c

```diff
@@ -100,7 +100,10 @@ int TDigestSketch_Create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc
             return REDISMODULE_ERR;
         }
     }
-    tdigest = td_new(compression);
```

Comment:
Why not leave the current behavior but check the pointer is not NULL?

Reply:
@ashtul the td_init() API will allow us to be more verbose on the error in the future if we want based on the reply value. If you agree lets use td_init instead of td_new. Furthermore td_new just simply calls td_init so it's one less function call.